### PR TITLE
Fix/167 bc header refactoring

### PR DESCRIPTION
**Pull request by mgaseta**
# Description

Refactored the Header component and fixed the pending minor issues.

Fixes #[167](https://github.com/bcgov/nr-spar/issues/167)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Ran unit tests and manual testing.

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend](https://nr-spar-178-backend.apps.silver.devops.gov.bc.ca/)
[Frontend](https://nr-spar-178-frontend.apps.silver.devops.gov.bc.ca/)
[Oracle-API](https://nr-spar-178-oracle-api.apps.silver.devops.gov.bc.ca/)

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge-main.yml)

### CODE REVIEW

File: frontend/src/components/BCHeader/styles.scss [hunk 1]
```diff
@@ -4,96 +4,107 @@
...
+    border-color: var(--#{vars.$bcgov-prefix}-icon-on-color);
   }
-}
-/* This rule is specific for the right panel only, it should fill the whole viewport
-   when the screen hits the panel size or smaller */
-@media only screen and (max-width: 400px) {
+
   .#{vars.$bcgov-prefix}--header-panel--expanded.notifications-panel,
   .#{vars.$bcgov-prefix}--header-panel--expanded.profile-panel {
+    z-index: 10000;
```
Comment (craigyu) [inline frontend/src/components/BCHeader/styles.scss:74]: [nice](https://youtu.be/SiMHTK15Pik?t=9)

**Review by craigyu — APPROVED** [review]
nice work! good to see disabled menu options

File: frontend/src/components/BCHeader/index.tsx [hunk 2]
```diff
@@ -174,74 +74,92 @@ const BCHeader = () => {
...
+              title={componentTexts.profile.title}
+              closeFn={() => closeRightPanel('myProfile')}
             />
             <MyProfile />
           </HeaderPanel>
-          <div className={overlay ? 'overlay-element active' : 'overlay-element'} />
-          <SideNav isChildOfHeader expanded={isSideNavExpanded} aria-label="Side menu">
+          {
+            overlay
+            && <div className="overlay-element" />
```
Comment (craigyu) [inline frontend/src/components/BCHeader/index.tsx:140]: This is a personal reference but I recently switched to using ternary operator for conditional rendering. 
I advocate that we all start using ternary as the standard practice, what do you think? 
https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx
https://stackoverflow.com/questions/65713434/react-render-logical-vs-ternary-operator

Comment (mgaseta) [reply frontend/src/components/BCHeader/index.tsx]: Yep, I agree, We don't lose anything perform related and it prevent errors and issues. This is [updated](https://github.com/bcgov/nr-spar/pull/178/commits/fee9adca3abd0c64f421685f825d5c27814824cb) now.

**Review by craigyu — APPROVED** [review]

**Pull request merged**